python_analytics/event_encoder.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, unicode_literals
+
+from abc import ABCMeta, abstractmethod
+from weakref import WeakKeyDictionary
+
+from six import add_metaclass, text_type
+
+
+NoValue = object()
+
+
+@add_metaclass(ABCMeta)
+class _TrackedAttribute(object):
```

Comment:
Google analytics calls these `Parameters`.

Maybe this should be `BaseParameter`. `TrackedAttribute` could become just `Parameter`. I don't like `EventParameter` for that because they have a specific class of event _called_ `event`, and others such as `pageview`. That would land up overloading the name `event` too hard.